cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -68,8 +73,37 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.MasterC
 	if err := dns.EnsureDNSAddon(cfg, client); err != nil {
 		errs = append(errs, err)
 	}
+
+	if err := removeDNSDeployment(cfg, client); err != nil {
+		errs = append(errs, err)
+	}
+
 	if err := proxy.EnsureProxyAddon(cfg, client); err != nil {
 		errs = append(errs, err)
 	}
 	return errors.NewAggregate(errs)
 }
+
+func removeDNSDeployment(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
+	if features.Enabled(cfg.FeatureGates, features.CoreDNS) {
+		ch := make(chan bool, 1)
+
+		go func() {
+			for {
+				getCoreDNS, _ := client.AppsV1beta2().Deployments(metav1.NamespaceSystem).Get("coredns", metav1.GetOptions{})
+				if getCoreDNS.Status.ReadyReplicas > 0 {
```

Comment:
You should not ignore the error here; check for the error and make sure `coreDNSDeployment` is non-nil